tests/test_kb_dataset.py

```diff
@@ -0,0 +1,127 @@
+            ids_seen = set()
+            for example in split:
+                self._assert_ids_globally_unique(example, ids_seen=ids_seen)
+
+    def test_do_all_referenced_ids_exist(self):
+        for split in self.dataset_bigbio.values():
+            for example in split:
+                referenced_ids = set()
+                existing_ids = set()
+
+                referenced_ids.update(self._get_referenced_ids(example))
+                existing_ids.update(self._get_existing_referable_ids(example))
+
+                for ref_id, ref_type in referenced_ids:
+                    if ref_type == "event_arg":
+                        self.assertTrue(
+                            (ref_id, "entity") in existing_ids
+                            or (ref_id, "event") in existing_ids
+                        )
+                    else:
+                        self.assertIn((ref_id, ref_type), existing_ids)
+
+    def _assert_ids_globally_unique(
+        self, collection, ids_seen: set, ignore_assertion: bool = False
+    ):
+        if isinstance(collection, dict):
+            for k, v in collection.items():
+                if isinstance(v, dict):
+                    self._assert_ids_globally_unique(v, ids_seen)
+                elif isinstance(v, list):
+                    for elem in v:
+                        self._assert_ids_globally_unique(elem, ids_seen)
+                else:
+                    if k == "id":
+                        if not ignore_assertion:
+                            self.assertNotIn(v, ids_seen)
+                        ids_seen.add(v)
+        elif isinstance(collection, list):
+            for elem in collection:
+                self._assert_ids_globally_unique(elem, ids_seen)
+
+    def _get_referenced_ids(self, example):
+        referenced_ids = []
+
+        for event in example["events"]:
+            for argument in event["arguments"]:
+                referenced_ids.append((argument["ref_id"], "event_arg"))
+
+        for coreference in example["coreferences"]:
+            for entity_id in coreference["entity_ids"]:
+                referenced_ids.append((entity_id, "entity"))
+
+        for relation in example["relations"]:
+            referenced_ids.append((relation["arg1_id"], "entity"))
+            referenced_ids.append((relation["arg2_id"], "entity"))
+
+        return referenced_ids
+
+    def _get_existing_referable_ids(self, example):
+        existing_ids = []
+
+        for entity in example["entities"]:
+            existing_ids.append((entity["id"], "entity"))
+
+        for event in example["events"]:
+            existing_ids.append((event["id"], "event"))
+
+        return existing_ids
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("datset_script")
+    args = parser.parse_args()
+
+    schema_file = Path(__file__).parent.parent / "schemas" / "kb.py"
+    # Load the schema
+    with open(schema_file) as f:
+        exec(f.read())
```

Comment:
can we make this an import somehow?

Reply:
yep, addressed with commit [0613907](https://github.com/bigscience-workshop/biomedical/pull/121/commits/0613907bcebe3c4ecc7f9295b89fbaa0d192b7d9)